gogs_client/interface.py

```diff
@@ -287,6 +343,271 @@ def delete_user(self, auth, username):
+        :raises ApiFailure: if the request cannot be serviced
+        """
+        path = "/repos/{u}/{r}/hooks/{i}".format(u=username, r=repo_name, i=hook_id)
+        response = self._check_ok(self._delete(path, auth=auth))
+        return response
+
+    def create_organization(self, auth, username, org_name, full_name=None, avatar_url=None, description=None, website=None, location=None):
+        """
+        Creates a new organization, and returns the created one.
+
+        :param auth.Authentication auth: authentication object, must be admin-level
+        :param str username: [Required] Organization user name
+        :param str full_name: Full name of organization 
+        :param str description: Description to the organization
+        :param str website: Official website
+        :param str location: Organization location
+        :return: a representation of the created organization
+        :rtype: GogsOrg
+        :raises NetworkFailure: if there is an error communicating with the server
+        :raises ApiFailure: if the request cannot be serviced
+        """
+
+        data = {
+          "username": org_name,
+          "full_name": full_name,
+          "description": description,
+          "website": website,
+          "location": location
+        }
+
+        url = "/admin/users/{u}/orgs".format(u=username) 
+        response = self._post(url, auth=auth, data=data)
+        self._check_ok(response)
+        return GogsOrg.from_json(response.json())
+
+    def create_organization_team(self, auth, org_name, name, description=None, permission="read"):
+        """
+        Creates a new team of the organization.
+
+        :param auth.Authentication auth: authentication object, must be admin-level
+        :param str org_name: [Required] Organization user name
+        :param str team_name: Full name of the team
+        :param str description: Description to the team
+        :param str permission: Team permission, can be read, write or admin, default is read
+        :return: a representation of the created team
+        :rtype: GogsOrg.Team
+        :raises NetworkFailure: if there is an error communicating with the server
+        :raises ApiFailure: if the request cannot be serviced
+        """
+
+        data = {
+          "name": name,
+          "description": description,
+          "permission": permission
+        }
+
+        url = "/admin/orgs/{o}/teams".format(o=org_name) 
+        response = self._post(url, auth=auth, data=data)
+        self._check_ok(response)
+        return GogsOrg.Team.from_json(response.json())
+
+    def add_team_membership(self, auth, team_id, username):
+        """
+        Add user to team.
+
+        :param auth.Authentication auth: authentication object, must be admin-level
+        :param str team_id: [Required] Id of the team
+        :param str username: Username of the user to be added to team
+        :return: status code of the request
```

Comment:
I don't really like returning the status code, since the point of the gogs_client is to abstract away details. I would either return the newly created team (if possible), or nothing.

Similarly for other methods below.